test/integration/puppetlabs/services/jruby/tasks_test.clj

```diff
@@ -21,33 +22,48 @@
    :metadata? schema/Bool
    :number-of-files (schema/pred #(<= % 5))})
 
-(schema/defn gen-task
-  "Assumes tasks dir already exists -- generates a set of task files for a
-  single task."
+(schema/defn gen-empty-task
+  "Assumes tasks dir already exists -- generates a set of task files for
+  a single task. All the generated payload files are empty, and all generated
+  metadata files contain the empty JSON object.
+
+  This function exists in addition to the tasks-generating utilities in
+  puppetlabs.puppetserver.testutils primarily because those other utilites only
+  generate things inside a hardcoded conf-dir at 'target/master-conf'."
   [env-dir :- (schema/cond-pre File schema/Str)
    task :- TaskOptions]
   (let [task-name (:name task)
         extensions '(".rb" "" ".sh" ".exe" ".py")
         task-dir (fs/file env-dir "modules" (:module-name task) "tasks")]
     (fs/mkdirs task-dir)
     (when (:metadata? task)
-      (fs/create (fs/file task-dir (str task-name ".json"))))
+      (spit (fs/file task-dir (str task-name ".json")) "{}"))
     (dotimes [n (:number-of-files task)]
       (fs/create (fs/file task-dir (str task-name (nth extensions n ".rb")))))))
 
-(defn gen-tasks
+(defn gen-empty-tasks
   "Tasks is a list of task maps, with keys:
   :name String, file name of task
   :module-name String, name of module task is in
   :metadata? Boolean, whether to generate a metadata file
-  :number-of-files Number, how many executable files to generate for the task (0 or more)"
+  :number-of-files Number, how many executable files to generate for the task (0 or more)
+
+  All generated files are empty, except metadata files, which contain the empty JSON object.
+
+  This function exists in addition to the tasks-generating utilities in
+  puppetlabs.puppetserver.testutils primarily because those other utilites only
+  generate things inside a hardcoded conf-dir at 'target/master-conf'."
   [env-dir tasks]
-  (dorun (map (partial gen-task env-dir) tasks)))
+  (dorun (map (partial gen-empty-task env-dir) tasks)))
 
 (defn create-env
   [env-dir tasks]
   (testutils/create-env-conf env-dir "")
-  (gen-tasks env-dir tasks))
+  (gen-empty-tasks env-dir tasks))
+
+(defn env-dir
```

Comment:
Does this actually get used? Looks like it's defined in a let block in the test. Did you mean to factor this out of the two tests?

Reply:
I updated the tests below to actually use this